lib/ocso_scraper/crawler.rb

```diff
@@ -0,0 +1,79 @@
+require 'httparty'
+require 'uri'
+require 'net/http'
+require 'openssl'
+
+module OcsoScraper
+  class Crawler
+    def self.perform
+      warrants = []
+      ('A'..'Z').each do |first_name_first_letter|
+        ('A'..'Z').each do |last_name_first_letter|
+          last_name_query = last_name_first_letter
+          next_last_name_query = nil
+          continue = true
+          while continue
+            results = request(first_name_first_letter, last_name_query)
+            page_warrants = PageParser.perform(results).sort_by! { |warrant| warrant[:last_name] }
+            warrants += page_warrants
+            if next_last_name_query.present?
+              last_name_query = next_last_name_query
+              next_last_name_query = nil
+            elsif page_warrants.count < page_limit
+              continue = false
+            else
+              last_name_second_letter = page_warrants.last[:last_name].chars[1]
```

Comment:
get second letter of last name from  the last records last name, then  adds the first letter to the 2nd and assigns it to last_name query. the next query is the same but for the second letter, I assume it goes to the next letter?( do not know what .succ does)

Reply:
.succ grabs the next letter in the alphabet